basic/conditional-area.js

```diff
@@ -0,0 +1,28 @@
+class Shape{
+  constructor (shape){
+    this.shape = shape;
+  }
+  area(arg)
+  {
+    
+    if(this.shape == 'circle'){
+      if(Object.keys(arg).length>1){
+        throw new Error("Wrong parameters for the given shape: Circle");
+      }
```

Comment:
You checked if the available `args` are greater than one.
If I tried to pass an argument like this:
```js
shape.area({width: 10 })
```
This will pass.

The best way to assert is to check if there is a key named `radius` in an object and only have one key. Otherwise, throw an error.

Reply:
**RESOLVED** 💯